gui/src/ts/components/pages/PathPage.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
+import {Frame} from "components/frame/Frame";
+import * as PathPageDocument from "api/queries/PathPageQuery.graphql";
+import {PathPageQuery} from "api/queries/types/PathPageQuery";
+import {useQuery} from "@apollo/react-hooks";
+import {ForceDirectedGraph} from "components/data/ForceDirectedGraph";
+import * as d3 from "d3";
+import {Node} from "models/Node";
+
+interface PathNode extends d3.SimulationNodeDatum, Node {
+  pathId: string;
+}
+
+interface PathEdge extends d3.SimulationLinkDatum<PathNode> {}
+
+export const PathPage: React.FunctionComponent = () => {
+  const {data} = useQuery<PathPageQuery>(PathPageDocument);
+  // console.dir(data);
+
+  const pathGraphData = React.useMemo<{
+    nodes: PathNode[];
+    links: PathEdge[];
+  }>(() => {
+    const nodes: PathNode[] = [];
+    const links: PathEdge[] = [];
+
+    data?.paths.forEach((path) => {
+      path.edges.forEach((edge) => {
+        if (
+          edge.subjectNode &&
```

Comment:
These should always exist but fine to check.

Reply:
Okay wasn't sure